aten/src/THC/THCStorage.cpp

```diff
@@ -60,25 +62,25 @@ THCStorage* THCStorage_newWithAllocator(THCState *state,
   return storage;
 }
 
-void THCStorage_free(THCState *state, THCStorage *self)
+void THCStorage_free(THCState *state, THCStorage *storage)
 {
-  AT_ASSERT(self->backend == at::kCUDA);
-
-  if(!(self->flag & TH_STORAGE_REFCOUNTED))
-    return;
+  AT_ASSERT(storage->backend == at::kCUDA);
 
-  if (--self->refcount == 0)
-  {
-    if(self->flag & TH_STORAGE_FREEMEM) {
-      auto* thc_device_allocator = static_cast<THCDeviceAllocator*>(self->allocatorVoidPtr);
-      THCudaCheck(
-        (*thc_device_allocator->free)(self->allocatorContext, self->data_ptr));
-    }
-    if(self->flag & TH_STORAGE_VIEW) {
-      THCStorage_free(state, self->view);
+  if ((storage->flag & TH_STORAGE_REFCOUNTED) && (storage->refcount.load() > 0)) {
```

Comment:
The `storage->refcount.load() > 0` checks feels wrong. If we're going to detect that we're in an inconsistent state, should throw an error or print a warning. 

I know this is from the THStorage_free version, so we can deal with it at some later point.

Reply:
Yeah, we should drop it. Also we should drop retainIfLive, it doesn't really have a place in this new world order.